test/QueryTest.hs

```diff
@@ -0,0 +1,60 @@
+module QueryTest where
+
+import            Control.Monad (void)
+import qualified  Data.ByteString as BS
+import            Data.List ((\\))
+import            Test.Tasty (TestTree, testGroup)
+import            Test.Tasty.HUnit (testCase, assertBool)
+
+import            Database.Kioku
+
+newtype TestData = TestData BS.ByteString
+  deriving (Eq, Ord, Show)
+
+instance Memorizable TestData where
+  memorize (TestData bytes) = bytes
+  recall = TestData
+
+testDataKey :: TestData -> BS.ByteString
+testDataKey (TestData bytes) = bytes
+
+test_queries :: TestTree
+test_queries =
+  testGroup
+    "Queries"
+    [ testCase "keyExactIn finds keys when TrieIndex breaks differently than query input" $ do
```

Comment:
This should really be three different `testCase`s so they can fail indpendently.